arch/riscv/boot/dts/bouffalolab/bl808.dtsi

```diff
@@ -97,13 +97,15 @@
 			status = "disabled";
 		};
 
-		ehci0: usb@20072000 {
-			compatible = "generic-ehci";
+		usb0: usb@20072000 {
+			compatible = "faraday,fotg210";
 			reg = <0x20072000 0x1000>;
 			interrupts-extended = <&ipclic BFLB_IPC_SOURCE_M0
 						       BFLB_IPC_DEVICE_USB
 						       IRQ_TYPE_EDGE_RISING>;
 			clocks = <&xtal>;
+			mboxes = <&ipclic BFLB_IPC_SOURCE_M0 BFLB_IPC_DEVICE_USB>;
+			dr_mode = "peripheral";
```

Comment:
Should this be "otg"? Does the driver have the ability to detect and switch modes?

Reply:
That will require some more coding at the usb-phy layer to sort out possible OTG operation. Right now the original driver assumes you either have 'host' or peripheral and then dispatches to either the hdc or udc driver. I think that might be able to be improved to OTG, but first I want to bring up the driver.

My primary motivation here is to get USB based networking and Serial gadgets working because that will dramatically improve my development time. Then focus on OTG and a proper physical layer.